### ISSUE TITLE
vpc_accept_aws_peering getting failed without waiting for timeout 

### ISSUE THREAD

**Issue by nvenkatapavan** [cloudamqp/terraform-provider-cloudamqp#171]
**Issue:** 
vpc_accept_aws_peering getting failed without waiting for time out. 

**Cloudamqp terraform module version:** 1.19.2

**Description:** 
After the Cloudamqp instance was created when cloudamqp_vpc_peering is running it's getting timeout within 10s, I also specified timeout to 3600 and sleep to 60 seconds, but still, it's failing within 10secs. The older version 1.9.2 version is working fine, but I see this issue is happening in the newer version. 

```
Error: waitForPeeringStatus failed, status: 204, message: map[]

  with module.test.cloudamqp_vpc_peering.vpc_accept_aws_peering["myapp"],
  on modules/aws-test/cloudamqp_vpc_peering.tf line 4, in resource "cloudamqp_vpc_peering" "vpc_accept_aws_peering":
   4: resource "cloudamqp_vpc_peering" "vpc_accept_aws_peering" {
```

Not sure if we are using the below code for Cloudamqp module, If so, Do we need to handle the 204 status code as well?
https://github.com/84codes/go-api/blob/master/api/vpc_peering.go#L23-L25

**Comment by nvenkatapavan** [comment]
Terraform code we are using : 
```
resource "cloudamqp_vpc_peering" "vpc_accept_aws_peering" {
  for_each = {
    for cloudamqp in local.cloudamqp_output : cloudamqp.app_name => cloudamqp
  }
  instance_id = each.value.instance_id
  peering_id  = aws_vpc_peering_connection.cloudamqp[each.key].id
  sleep = 60
  timeout = 3600 
  depends_on = [
    aws_vpc_peering_connection.cloudamqp,
    module.cloudamqp
  ]
}
```

**Comment by dentarg** [comment]
Hi, what's happening is that the API returns a 204 response when the peering ID isn't correct (can't be found in the list of peerings for your VPC).

**Comment by dentarg** [comment]
We will add a task to our backlog to make the error more clear.

**Comment by nvenkatapavan** [comment]
@dentarg Thanks for the quick response. Actually Same code is working in 1.9.2 version. Let me check terraform code again if something i'm missing or any changes required for newer code.

**Comment by dentarg** [comment]
Okay, that's strange... terraform-provider-cloudamqp v1.9.2 is using go-api v1.3.4: https://github.com/84codes/go-api/blob/v1.3.4/api/vpc_peering.go#L9-L29 the only difference I can see compared to the latest version (https://github.com/84codes/go-api/blob/v1.9.1/api/vpc_peering.go#L10-L31) is that the sleep call comes first now, rather than last

**Comment by dentarg** [comment]
@nvenkatapavan If you can email support@cloudamqp.com with details about your account/cluster info we can take a deeper look

**Comment by nvenkatapavan** [comment]
@dentarg I sent email on this issue. Let me add cluster details and more logs.

**Comment by nvenkatapavan** [comment]
@dentarg I sent an email from Pavan.Kumar.N.V@kyndryl.com with cluster details and logs. Please let me know if anything needed from my end. I will provide it.

**Comment by dentarg** [comment]
Thanks we got it, we will look into the issues

**Comment by dentarg** [comment]
> We will add a task to our backlog to make the error more clear.

We will also make updates to the provider to retry this error, we will leave this issue open until we have added that.

**Comment by tbroden84** [comment]
@nvenkatapavan Released a new version [v1.19.3](https://github.com/cloudamqp/terraform-provider-cloudamqp/releases/tag/v1.19.3) that includes waiting for AWS peering to appear during the resource timeout.

**Comment by nvenkatapavan** [comment]
Thank you so much for the fix @tbroden84. Let me try new version.
